apps/web/app/features/stacking/utils/utils-allow-contract-caller.ts

```diff
@@ -21,41 +24,36 @@ function getOptions(
   };
 }
 
-export interface HandleAllowContractCallerArgs {
-  poxWrapperContract: WrapperPrincipal;
-  onFinish: () => void;
+export function isAllowContractCallerConfirmed(
+  poolName: PoolName,
+  network: StacksNetwork,
+  hasUserConfirmedPoolWrapperContract: PoolWrapperAllowanceState
+): boolean {
+  const poxWrapperContract = getPoxWrapperContract(poolName, network);
+  const networkInstance = getNetworkInstance(network);
+
+  return Boolean(hasUserConfirmedPoolWrapperContract[networkInstance]?.[poxWrapperContract]);
 }
+
 interface CreateHandleSubmitArgs {
   client: StackingClient;
   network: StacksNetworkName;
-  setIsContractCallExtensionPageOpen: Dispatch<SetStateAction<boolean>>;
+  poxWrapperContract: WrapperPrincipal;
 }
-export function createHandleSubmit({
+
+export function createAllowContractCallerSubmitMutationOptions({
   client,
   network,
-  setIsContractCallExtensionPageOpen,
+  poxWrapperContract,
 }: CreateHandleSubmitArgs) {
-  return async function handleSubmit({
-    poxWrapperContract,
-    onFinish,
-  }: HandleAllowContractCallerArgs) {
-    // TODO: handle thrown errors
-    const [stackingContract] = await Promise.all([client.getStackingContract()]);
-
-    const allowContractCallerOptions = getOptions(poxWrapperContract, stackingContract, network);
-
-    setIsContractCallExtensionPageOpen(true);
-
-    try {
-      await leather.stxCallContract({
-        ...allowContractCallerOptions,
-      });
-      onFinish?.();
-    } catch (error) {
-      // eslint-disable-next-line no-console
-      console.error(error);
-    } finally {
-      setIsContractCallExtensionPageOpen(false);
-    }
-  };
+  return {
+    mutationKey: ['allowContractCaller', client, network, poxWrapperContract],
+    mutationFn: async () => {
+      const [stackingContract] = await Promise.all([client.getStackingContract()]);
+
+      const allowContractCallerOptions = getOptions(poxWrapperContract, stackingContract, network);
+
+      return leather.stxCallContract(allowContractCallerOptions);
```

Comment:
How come leather call is within the mutateFn?

Reply:
what's wrong with it?

Reply:
Not inherently suggesting something is wrong with it, but I'm trying to understand why it's there, as typically it's used when server data is mutated?

Reply:
we mutate blockchain data here, no? 😁
e.g. wagmi useWriteContract uses useMutation under the hood https://wagmi.sh/react/api/hooks/useWriteContract